myst_nbs/case_studies/regression_discontinuity.myst.md

```diff
@@ -0,0 +1,219 @@
+- We assume we have accurately observed whether a unit has been treated or not. That is, this model assumes a sharp discontinuity with no uncertainty.
+
+```{code-cell} ipython3
+with pm.Model() as model:
+    x = pm.MutableData("x", df.x, dims="obs_id")
+    treated = pm.MutableData("treated", df.treated, dims="obs_id")
+    sigma = pm.HalfNormal("sigma", 1)
+    delta = pm.Cauchy("effect", alpha=0, beta=1)
+    mu = pm.Deterministic("mu", x + (delta * treated), dims="obs_id")
+    pm.Normal("y", mu=mu, sigma=sigma, observed=df.y, dims="obs_id")
+
+pm.model_to_graphviz(model)
+```
+
+## Inference
+
+```{code-cell} ipython3
+with model:
+    idata = pm.sample(random_seed=123)
+```
+
+We can see that we get no sampling warnings, and that plotting the MCMC traces shows no issues.
+
+```{code-cell} ipython3
+az.plot_trace(idata, var_names=["effect", "sigma"]);
+```
+
+We can also see that we are able to accurately recover the true discontinuity magnitude (left) and the standard deviation of the change in units between pre- and post-test (right).
+
+```{code-cell} ipython3
+az.plot_posterior(idata, var_names=["effect", "sigma"], ref_val=[treatment_effect, sd]);
+```
+
+## Counterfactual questions
+
+We can use posterior prediction to ask what would we expect to see if:
+- no units were exposed to the treatment (blue shaded region)
+- all units were exposed to the treatment (orange shaded region).
+
+```{code-cell} ipython3
+:tags: [hide-input]
+
+# MODEL EXPECTATION WITHOUT TREATMENT ------------------------------------
+# probe data
+_x = np.linspace(np.min(df.x), np.max(df.x), 500)
+_treated = np.zeros(_x.shape)
+
+# posterior prediction
+with model:
+    pm.set_data({"x": _x, "treated": _treated})
+    ppc = pm.sample_posterior_predictive(idata, var_names=["mu", "y"])
+
+# plotting
+ax = plot_data(df)
+_y = ppc.posterior_predictive.mu.mean(dim=["chain", "draw"])
+az.plot_hdi(_x, ppc.posterior_predictive["mu"], color="C0", hdi_prob=0.95)
+
+# MODEL EXPECTATION WITH TREATMENT ---------------------------------------
+# probe data
+_x = np.linspace(np.min(df.x), np.max(df.x), 500)
+_treated = np.ones(_x.shape)
+
+# posterior prediction
+with model:
+    pm.set_data({"x": _x, "treated": _treated})
+    ppc = pm.sample_posterior_predictive(idata, var_names=["mu", "y"])
+
+# plotting
+_y = ppc.posterior_predictive.mu.mean(dim=["chain", "draw"])
+az.plot_hdi(_x, ppc.posterior_predictive["mu"], color="C1", hdi_prob=0.95)
+plt.legend();
+```
+
+The blue shaded region (which is very narrow) shows the 95% credible region of the expected value of the post-test measurement for a range of possible pre-test measures. This is actually very interesting because it is an example of counterfactual inference. We did not observe any units that were untreated above the threshold. But assuming our model is a good description of reality, we can ask the counterfactual question of "what if a unit above the threshold was not treated?"
```

Comment:
> shows the 95% credible region of the **expected value of the post-test measurement** for a range of possible pre-test measures

This is not right, the plot is on `mu` which is the mean of the measurements, not the actual measurements. Otherwise the model would be way off in calibration terms. When plotting the 95% region of the measurements _for the same `x` as the observations_ roughly 95% of the observations should fall inside that region. Note the emphasis becasue this applies x-wise and might not be true for the whole plotted region with ArviZ defaults because of the smoothing.

Reply:
I thought this is what I was saying by "95% credible region of the **expected value** of the post-test measurement"

Maybe it would be clearer by just deleting "measurement"?

Reply:
Riight, sorry about that,  "expected value" is a synonym of first moment. I always forget this.

Removing measurement might help yes, but I am not sure will be enough. I find this confusing for two reasons.

We are defining a new random variable E(measurement) and giving confidence intervals on that, instead of doing it on the actual measurements which are the terms in which most people think. I am not sure using this new variable instead of y directly helps illustrating the point that comes later, which if I understand correctly is about raw values. 

"expected value" doesn't seem like a technical term. I always get it wrong and I think it happens to many other non native speakers as confusion on credible regions of a value and of its mean are common questions I get. Using expectation or expectancy for example might help on that end.